docs/proposals/docker-registry-split.md

```diff
@@ -0,0 +1,141 @@
+# Move docker-registry to separate repository
+
+## Problem
+
+The `docker/distribution` has a lot of dependencies. Since `kubernetes` began to use docker/distribution
+it has become very difficult to update `docker/distribution` in the part where you need to combine the
+dependencies of these projects. There is no guarantee that the merge of the dependencies would not make
+regression in these projects.
+
+## Constraints and Assumptions
+
+The docker-registry is highly integrated in a common code-base so it is very difficult to describe
+all the difficulties of separation.
+
+## Investigation
+
+### etcd
+
+The docker-registry does not have its own `etcd`. All metadata are stored to the `origin` database using
```

Comment:
worth to mention the problem with storage if we continue using the openshift etcd.